Adamant/Modules/Chat/ViewModel/ChatViewModel.swift

```diff
@@ -610,8 +722,258 @@ final class ChatViewModel: NSObject {
+        else { return }
+        
+        let chatFiles = fileModel.value.content.fileModel.files
+        
+        let isPreviewAutoDownloadAllowed = isDownloadAllowed(
+            policy: filesStorageProprieties.autoDownloadPreviewPolicy(),
+            havePartnerName: havePartnerName
+        )
+        
+        if !isPreviewAutoDownloadAllowed,
+           file.previewImage == nil,
+           file.file.preview != nil,
+           !chatFileService.isDownloadPreviewLimitReached(for: file.file.id) {
+            forceDownloadAllFiles(messageId: messageId, files: chatFiles)
+            return
+        }
+        
+        guard !file.isCached,
+              !filesStorage.isCachedLocally(file.file.id)
+        else {
+            Task {
+                self.presentFileInFullScreen(id: file.file.id, chatFiles: chatFiles)
+            }
+            return
+        }
+        
+        guard tx?.statusEnum == .delivered else { return }
+        
+        downloadFile(
+            file: file,
+            previewDownloadAllowed: true,
+            fullMediaDownloadAllowed: true
+        )
+    }
+    
+    func downloadContentIfNeeded(
+        messageId: String,
+        files: [ChatFile]
+    ) {
+        let tx = chatTransactions.first(where: { $0.txId == messageId })
+        
+        guard tx?.statusEnum == .delivered || tx?.statusEnum == nil else { return }
+        
+        let chatFiles = files.filter {
+            $0.fileType == .image || $0.fileType == .video
+        }
+        
+        chatFiles.forEach { file in
+            Task {
+                await chatFileService.autoDownload(
+                    file: file,
+                    chatroom: chatroom,
+                    havePartnerName: havePartnerName,
+                    previewDownloadPolicy: filesStorageProprieties.autoDownloadPreviewPolicy(),
+                    fullMediaDownloadPolicy: filesStorageProprieties.autoDownloadFullMediaPolicy(),
+                    saveEncrypted: filesStorageProprieties.saveFileEncrypted()
+                )
+            }
+        }
+    }
+    
+    func forceDownloadAllFiles(messageId: String, files: [ChatFile]) {
+        let isPreviewDownloadAllowed = isDownloadAllowed(
+            policy: filesStorageProprieties.autoDownloadPreviewPolicy(),
+            havePartnerName: havePartnerName
+        )
+        
+        let isFullMediaDownloadAllowed = isDownloadAllowed(
+            policy: filesStorageProprieties.autoDownloadFullMediaPolicy(),
+            havePartnerName: havePartnerName
+        )
+        
+        let needToDownload: [ChatFile]
+        
+        let shouldDownloadFile: (ChatFile) -> Bool = { file in
+            if !file.isCached {
+                return true
+            }
+            
+            if file.fileType.isMedia && file.previewImage == nil {
+                return isPreviewDownloadAllowed
+            }
+            
+            return false
```

Comment:
```
if !file.isCached {
                 return true
             }

             if file.fileType.isMedia && file.previewImage == nil {
                 return isPreviewDownloadAllowed
             }

             return false
```
Always simplify if possible
`!file.isCached || file.fileType.isMedia && file.previewImage == nil && isPreviewDownloadAllowed`
(check pls if it's correct before inserting)